geomstats/geometry/pullback_metric.py

```diff
@@ -138,20 +139,24 @@ def immersion_a(x):
             )
             if self.dim == 1 and hessian_a.ndim > 2:
                 hessian_a = gs.squeeze(hessian_a, axis=-1)
+            elif self.dim != 1 and hessian_a.ndim > 2:
+                hessian_a = gs.squeeze(hessian_a)
+
```

Comment:
This if is not immediately obvious, maybe this instead?

```python

if hessian_a.ndim > 2:
  axis = -1 if self.dim ==1 else None
  hessian_a = gs.squeeze(hessian_a, axis=axis)
```

btw, the use of `squeeze` is not obvious. maybe a good place to drop a comment?

same for jacobian.

Reply:
Good point, I will look at this in the other open PR on pullback metrics: it will be better so separate the two contributions, and avoid github conflicts.